extensions/http-curl/tests/unit/HTTPStreamingCallbackTests.cpp

```diff
@@ -0,0 +1,159 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <thread>
+#include <mutex>
+#include <vector>
+#include <string>
+#include <chrono>
+#include <cstring>
+#include <cstdint>
+
+#include "client/HTTPCallback.h"
+#include "TestBase.h"
+
+class HttpStreamingCallbackTestsFixture {
+ public:
+  HttpStreamingCallbackTestsFixture() {
+    LogTestController::getInstance().setTrace<utils::HttpStreamingCallback>();
+  }
+
+  virtual ~HttpStreamingCallbackTestsFixture() {
+    if (consumer_thread_.joinable()) {
+      consumer_thread_.join();
+    }
+    LogTestController::getInstance().reset();
+  }
+
+  void startConsumerThread() {
+    if (consumer_thread_.joinable()) {
+      throw std::logic_error("Consumer thread already started");
+    }
+    consumer_thread_ = std::thread([this](){
+      std::cerr << "Consumer thread started" << std::endl;
+
+      size_t current_pos = 0U;
```

Comment:
If I understand correctly, `current_pos == content_.size()`, except between lines 62,65. Could we reduce the scope of this variable to the scope where the `memcpy` happens?

Reply:
- This function was made to emulate `utils::HTTPCLient::send_write` closely, and the position there is a member variable, that's why `current_pos` became a variable in this scope
 - I couldn't possibly care less about the scope of a variable inside a lambda in a test function